packages/abi-gen/test/generate_test/test/abi_gen_dummy_test.ts

```diff
@@ -0,0 +1,130 @@
+            const result = await abiGenDummy.simplePureFunction.callAsync();
+            expect(result).bignumber.to.equal(new BigNumber(1));
+        });
+    });
+    describe('simplePureFunctionWithInput', () => {
+        it('should call simplePureFunctionWithInput', async () => {
+            const result = await abiGenDummy.simplePureFunctionWithInput.callAsync(new BigNumber(5));
+            expect(result).bignumber.to.equal(new BigNumber(6));
+        });
+    });
+    describe('pureFunctionWithConstant', () => {
+        it('should call pureFunctionWithConstant', async () => {
+            const result = await abiGenDummy.pureFunctionWithConstant.callAsync();
+            expect(result).bignumber.to.equal(new BigNumber(1234));
+        });
+    });
+    describe('simpleRevert', () => {
+        it('should call simpleRevert', async () => {
+            return expectContractCallFailedAsync(abiGenDummy.simpleRevert.callAsync(), RevertReason.ValidatorError);
+        });
+    });
+    describe('revertWithConstant', () => {
+        it('should call revertWithConstant', async () => {
+            return expectContractCallFailedAsync(
+                abiGenDummy.revertWithConstant.callAsync(),
+                RevertReason.ValidatorError,
+            );
+        });
+    });
+    describe('simpleRequire', () => {
+        it('should call simpleRequire', async () => {
+            return expectContractCallFailedAsync(abiGenDummy.simpleRequire.callAsync(), RevertReason.ValidatorError);
+        });
+    });
+    describe('requireWithConstant', () => {
+        it('should call requireWithConstant', async () => {
+            return expectContractCallFailedAsync(
+                abiGenDummy.requireWithConstant.callAsync(),
+                RevertReason.ValidatorError,
+            );
+        });
+    });
+});
+
+describe('Lib dummy contract', () => {
+    let libDummy: TestLibDummyContract;
+    before(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    after(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    before(async () => {
+        libDummy = await TestLibDummyContract.deployFrom0xArtifactAsync(artifacts.TestLibDummy, provider, txDefaults);
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+
+    it('should call a library function', async () => {
+        const result = await libDummy.publicAddOne.callAsync(new BigNumber(1));
+        expect(result).bignumber.to.equal(new BigNumber(2));
+    });
+
+    it('should call a library function referencing a constant', async () => {
+        const result = await libDummy.publicAddConstant.callAsync(new BigNumber(1));
+        expect(result).bignumber.to.equal(new BigNumber(1235));
+    });
+});
+
+// HACK (xianny): copied from @0x/contracts-test-utils to avoid circular dependency
```

Comment:
```suggestion
// HACK(xianny): copied from @0x/contracts-test-utils to avoid circular dependency
```

Reply:
Is this causing a circular dep event when imported as a `devDependency`?

Reply:
Yeah, setting it to `devDependency` instead of `dependency` doesn't seem to make a difference :/ 

```
ERROR: Dependency cycle detected:
   @0x/abi-gen-wrappers <- @0x/abi-gen <- @0x/contracts-test-utils <- @0x/order-utils <- @0x/abi-gen-wrappers 
```

I removed `abi-gen` completely from `contracts-test-utils`. This is with `abi-gen` listed as `devDependency` in `abi-gen-wrappers`.